src/features/user-dashboard/components/UserDetailForm.tsx

```diff
@@ -21,23 +22,28 @@ export interface Props {
   serverValidationErrors: ServerValidationErrors<FormData> | null;
 }
 
-const schema = yup.object({
-  firstName: yup.string().required('First Name is required.'),
-  lastName: yup.string().required('Last Name is required.'),
-  email: yup.string().email().required('Email is required.'),
-  role: yup.string().required('Role is required.'),
-});
-
 export const UserDetailForm: FC<Props> = ({
   availableRoles,
   defaultValues = {},
   onSubmit,
   submitButtonLabel = 'Submit',
   serverValidationErrors,
 }) => {
+  const { t } = useTranslation(['translation', 'common']);
+
   const options: RoleOption[] = [];
   availableRoles.forEach(role => {
-    options.push({ label: role, value: role });
+    options.push({ label: t(role.toLocaleLowerCase(), { ns: 'common' }), value: role });
```

Comment:
I'm using the role name as a key here to grab the translated role name. If new roles were added to the application, the developer would also need to add the appropriate translation strings.